qiskit/optimization/algorithms/admm_optimizer.py

```diff
@@ -509,22 +576,24 @@ def _create_step1_problem(self) -> QuadraticProgram:
         """
         op1 = QuadraticProgram()
 
-        binary_size = len(self._state.binary_indices)
+        binary_size = len(self._state.step1_absolute_indices)
         # create the same binary variables.
         for i in range(binary_size):
-            op1.binary_var(name="x0_" + str(i + 1))
+            name = self._state.op.variables[self._state.step1_absolute_indices[i]].name
+            op1.binary_var(name=name)
 
         # prepare and set quadratic objective.
-        quadratic_objective = self._state.q0 +\
+        quadratic_objective = self._state.q0 + \
             self._params.factor_c / 2 * np.dot(self._state.a0.transpose(), self._state.a0) +\
             self._state.rho / 2 * np.eye(binary_size)
         op1.objective.quadratic = quadratic_objective
 
         # prepare and set linear objective.
         linear_objective = self._state.c0 - \
-            self._params.factor_c * np.dot(self._state.b0, self._state.a0) + \
-            self._state.rho * (- self._state.y - self._state.z) + \
-            self._state.lambda_mult
+                           self._params.factor_c * np.dot(self._state.b0, self._state.a0) + \
+                           self._state.rho * (- self._state.y[self._state.step1_relative_indices] -
+                                              self._state.z[self._state.step1_relative_indices]) + \
+                           self._state.lambda_mult[self._state.step1_relative_indices]
```

Comment:
Style is failing these lines as over indented. You can see the prior formatting above which passes. If you want to check locally you can run `make style` locally, also `make lint` etc. See the CONTRIBUTING.md file for more info.

Reply:
Thanks, already noticed and fixed.